<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Weiyue Ding" />

<meta name="date" content="2024-08-17" />

<title>sintHiChIP</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
span.underline{text-decoration: underline;}
div.column{display: inline-block; vertical-align: top; width: 50%;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
</style>



<style type="text/css">
code {
white-space: pre;
}
.sourceCode {
overflow: visible;
}
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
{ counter-reset: source-line 0; }
pre.numberSource code > span
{ position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
{ content: counter(source-line);
position: relative; left: -1em; text-align: right; vertical-align: baseline;
border: none; display: inline-block;
-webkit-touch-callout: none; -webkit-user-select: none;
-khtml-user-select: none; -moz-user-select: none;
-ms-user-select: none; user-select: none;
padding: 0 4px; width: 4em;
color: #aaaaaa;
}
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa; padding-left: 4px; }
div.sourceCode
{ }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } 
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.at { color: #7d9029; } 
code span.bn { color: #40a070; } 
code span.bu { color: #008000; } 
code span.cf { color: #007020; font-weight: bold; } 
code span.ch { color: #4070a0; } 
code span.cn { color: #880000; } 
code span.co { color: #60a0b0; font-style: italic; } 
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.do { color: #ba2121; font-style: italic; } 
code span.dt { color: #902000; } 
code span.dv { color: #40a070; } 
code span.er { color: #ff0000; font-weight: bold; } 
code span.ex { } 
code span.fl { color: #40a070; } 
code span.fu { color: #06287e; } 
code span.im { color: #008000; font-weight: bold; } 
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } 
code span.kw { color: #007020; font-weight: bold; } 
code span.op { color: #666666; } 
code span.ot { color: #007020; } 
code span.pp { color: #bc7a00; } 
code span.sc { color: #4070a0; } 
code span.ss { color: #bb6688; } 
code span.st { color: #4070a0; } 
code span.va { color: #19177c; } 
code span.vs { color: #4070a0; } 
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } 
</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">sintHiChIP</h1>
<h4 class="author">Weiyue Ding</h4>
<h4 class="date">2024-08-17</h4>



<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p>sintHiChIP is a comprehensive R package designed for the analysis of
HiChIP data, offering both local and global modes of operation. By
correcting the bias of restriction enzyme cut site density, sintHiChIP
facilitates the identification and characterization of significant
chromatin interactions from HiChIP experiments.</p>
</div>
<div id="platform" class="section level2">
<h2>Platform</h2>
<p>sintHiChIP is primarily designed to run on Linux and Unix-like
operating systems. This includes various Linux distributions (such as
Ubuntu, CentOS, Fedora) as well as Unix-based systems like macOS. While
it may work on other platforms, we strongly recommend using a Linux or
Unix environment for optimal performance and compatibility.</p>
</div>
<div id="prerequisites" class="section level2">
<h2>Prerequisites</h2>
<p>Before installing and using sintHiChIP, ensure you have the following
software and R packages installed. Pay special attention to the
requirements for local mode.</p>
<div id="required-software" class="section level3">
<h3>Required Software</h3>
<ol style="list-style-type: decimal">
<li>R (version 4.0.0 or higher)</li>
<li>Python (version 3.6 or higher)</li>
</ol>
</div>
<div id="for-local-mode" class="section level3">
<h3>For Local Mode:</h3>
<ol start="3" style="list-style-type: decimal">
<li>hichipper (version 0.7.7 or higher)
<ul>
<li>Used for peak calling and interaction identification in local
mode</li>
<li>Installation: <code>pip install hichipper</code></li>
</ul></li>
<li>bedtools (version 2.29.0 or higher)
<ul>
<li>Used for genomic interval manipulations</li>
<li>Installation: <code>sudo apt-get install bedtools</code>
(Ubuntu/Debian)</li>
</ul></li>
<li>tabix (usually comes with samtools, version 1.10 or higher)
<ul>
<li>Used for indexing and querying TAB-delimited genome position
files</li>
<li>Installation: <code>sudo apt-get install tabix</code>
(Ubuntu/Debian)</li>
</ul></li>
<li>bgzip (usually comes with samtools, version 1.10 or higher)
<ul>
<li>Used for blocking compression of genomic data files</li>
<li>Installation: Comes with tabix</li>
</ul></li>
</ol>
</div>
<div id="input-data-requirements" class="section level3">
<h3>Input Data Requirements</h3>
<p>sintHiChIP requires pre-processed HiChIP data. This data should be in
a format compatible with the tool, typically including:</p>
<ol style="list-style-type: decimal">
<li>For Local Mode:
<ul>
<li>Aligned and filtered read pairs output folder (often the output of
tools like HiC-Pro or similar pipelines)</li>
<li>Peak files (if not using hichipper for peak calling)</li>
<li>Restriction fragment information</li>
<li>Normalization restriction enzyme cut site density file</li>
<li>the build_matrix file from HiC-Pro utilites</li>
</ul></li>
<li>For Global Mode:
<ul>
<li>Valid pairs file</li>
<li>Chromosome size file</li>
<li>Peak file</li>
<li>Normalizated restriction enzyme cut site density file</li>
</ul></li>
</ol>
<p>Note: While sintHiChIP does not require HiC-Pro to be installed, it
does expect input data in a format similar to HiC-Pro output. Users
should ensure their data is properly formatted regardless of the
preprocessing pipeline used.</p>
</div>
<div id="required-r-packages" class="section level3">
<h3>Required R Packages</h3>
<p>You can install Bioconductor and the required R packages with:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="co"># Ensure BiocManager is installed</span></span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;BiocManager&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>)) {</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a>    <span class="fu">install.packages</span>(<span class="st">&quot;BiocManager&quot;</span>)</span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a>}</span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a></span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a><span class="co"># Install Bioconductor packages if not already installed</span></span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a>bioc_pkgs <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;GenomicRanges&quot;</span>, <span class="st">&quot;IRanges&quot;</span>, <span class="st">&quot;S4Vectors&quot;</span>, <span class="st">&quot;GenomeInfoDb&quot;</span>)</span>
<span id="cb1-8"><a href="#cb1-8" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(bioc_pkgs[<span class="sc">!</span>bioc_pkgs <span class="sc">%in%</span> <span class="fu">installed.packages</span>()[,<span class="st">&quot;Package&quot;</span>]])</span>
<span id="cb1-9"><a href="#cb1-9" tabindex="-1"></a></span>
<span id="cb1-10"><a href="#cb1-10" tabindex="-1"></a><span class="co"># Install CRAN packages if not already installed</span></span>
<span id="cb1-11"><a href="#cb1-11" tabindex="-1"></a>cran_pkgs <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;data.table&quot;</span>, <span class="st">&quot;dplyr&quot;</span>, <span class="st">&quot;readr&quot;</span>, <span class="st">&quot;yaml&quot;</span>, <span class="st">&quot;Rcpp&quot;</span>, <span class="st">&quot;matrixStats&quot;</span>, <span class="st">&quot;testthat&quot;</span>, <span class="st">&quot;knitr&quot;</span>, <span class="st">&quot;rmarkdown&quot;</span>)</span>
<span id="cb1-12"><a href="#cb1-12" tabindex="-1"></a><span class="fu">install.packages</span>(cran_pkgs[<span class="sc">!</span>cran_pkgs <span class="sc">%in%</span> <span class="fu">installed.packages</span>()[,<span class="st">&quot;Package&quot;</span>]])</span></code></pre></div>
</div>
</div>
<div id="installation" class="section level2">
<h2>Installation</h2>
<p>You can install sintHiChIP directly from GitHub using the devtools
package:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;devtools&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>))</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a>    <span class="fu">install.packages</span>(<span class="st">&quot;devtools&quot;</span>)</span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a></span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a>devtools<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">&quot;YourGitHubUsername/sintHiChIP&quot;</span>)</span></code></pre></div>
</div>
<div id="usage" class="section level2">
<h2>Usage</h2>
<p>sintHiChIP can be run in two modes: local and global. Each mode has
its specific parameters and use cases.</p>
</div>
<div id="local-mode" class="section level2">
<h2>Local Mode</h2>
<p>Local mode is designed for analyzing specific regions of interest
(Peak-to-Peak). It uses either hichipper for peak calling (default
“EACH, ALL”), or the input ChIP-seq or HiChIP peaks to identify local
interaction.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="fu">library</span>(sintHiChIP)</span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a></span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a><span class="fu">run_sintHiChIP</span>(</span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a>  <span class="at">mode =</span> <span class="st">&quot;local&quot;</span>,</span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a>  <span class="at">outdir =</span> <span class="st">&quot;P2P&quot;</span>,</span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a>  <span class="at">yaml =</span> <span class="st">&quot;config.yaml&quot;</span>,</span>
<span id="cb3-7"><a href="#cb3-7" tabindex="-1"></a>  <span class="at">peaks =</span> <span class="st">&quot;EACH,ALL&quot;</span>,</span>
<span id="cb3-8"><a href="#cb3-8" tabindex="-1"></a>  <span class="at">resfrags =</span> <span class="st">&quot;path/to/restriction_fragments.bed&quot;</span>,</span>
<span id="cb3-9"><a href="#cb3-9" tabindex="-1"></a>  <span class="at">hicpro_output =</span> <span class="st">&quot;path/to/hicpro_results&quot;</span>,</span>
<span id="cb3-10"><a href="#cb3-10" tabindex="-1"></a>  <span class="at">normSiteFile =</span> <span class="st">&quot;path/to/norm_sites.bed&quot;</span>,</span>
<span id="cb3-11"><a href="#cb3-11" tabindex="-1"></a>  <span class="at">FDR =</span> <span class="fl">0.01</span></span>
<span id="cb3-12"><a href="#cb3-12" tabindex="-1"></a>)</span></code></pre></div>
<div id="local-mode-parameters" class="section level3">
<h3>Local Mode Parameters</h3>
<p>Please choose the significant parameters below when using local
mode:</p>
<ul>
<li>mode: Set to “local” for local mode.</li>
<li>outdir: Output directory name (folder name only, will be created in
the same directory as the YAML file).</li>
<li>yaml: Path to the YAML configuration file.</li>
<li>peaks: hichipper peaks options (default “EACH,ALL”) or path to peak
file.</li>
<li>resfrags: Path to restriction fragments file.</li>
<li>hicpro_output: Path to HiC-Pro output directory.</li>
<li>keep_temp: Whether to keep temporary files (default is FALSE).</li>
<li>normSiteFile: Path to normalization restriction enzyme cut site
density file.</li>
<li>FDR: False Discovery Rate threshold for significant HiChIP loops
(default 0.01).</li>
</ul>
</div>
</div>
<div id="global-mode" class="section level2">
<h2>Global Mode</h2>
<p>Global mode is used for genome-wide analysis. It starts from valid
pairs and performs binning and interaction calling.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="fu">run_sintHiChIP</span>(</span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a>  <span class="at">mode =</span> <span class="st">&quot;global&quot;</span>,</span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a>  <span class="at">outdir =</span> <span class="st">&quot;/path/to/output&quot;</span>,</span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a>  <span class="at">valid_pairs =</span> <span class="st">&quot;/path/to/valid_pairs.txt&quot;</span>,</span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a>  <span class="at">chr_size =</span> <span class="st">&quot;/path/to/chrom_sizes.txt&quot;</span>,</span>
<span id="cb4-6"><a href="#cb4-6" tabindex="-1"></a>  <span class="at">build_matrix =</span> <span class="st">&quot;/path/to/build_matrix&quot;</span>,</span>
<span id="cb4-7"><a href="#cb4-7" tabindex="-1"></a>  <span class="at">peaks =</span> <span class="st">&quot;/path/to/peaks.bed&quot;</span>,</span>
<span id="cb4-8"><a href="#cb4-8" tabindex="-1"></a>  <span class="at">normSiteFile =</span> <span class="st">&quot;/path/to/norm_sites.bed&quot;</span>,</span>
<span id="cb4-9"><a href="#cb4-9" tabindex="-1"></a>  <span class="at">bin_size =</span> <span class="dv">5000</span>,</span>
<span id="cb4-10"><a href="#cb4-10" tabindex="-1"></a>  <span class="at">FDR =</span> <span class="fl">0.01</span></span>
<span id="cb4-11"><a href="#cb4-11" tabindex="-1"></a>)</span></code></pre></div>
<div id="global-mode-parameters" class="section level3">
<h3>Global Mode Parameters</h3>
<p>Please choose the significant parameters below when using global
mode:</p>
<ul>
<li>mode: Set to “global” for global mode.</li>
<li>outdir: Full path to output directory.</li>
<li>valid_pairs: Path to valid pairs file.</li>
<li>chr_size: Path to chromosome size file.</li>
<li>build_matrix: Path to build matrix tool.</li>
<li>peaks: Path to peak file.</li>
<li>bin_size: Bin size for matrix generation (default 5000).</li>
<li>normSiteFile: Path to normalization restriction enzyme cut site
density file.</li>
<li>FDR: False Discovery Rate threshold for significant HiChIP loops
(default 0.01).</li>
</ul>
</div>
</div>
<div id="output-files" class="section level2">
<h2>Output Files</h2>
<p>sintHiChIP generates several significant output files:</p>
<ol style="list-style-type: decimal">
<li>Filtered interaction files (*.all.intra.loop_counts.bedpe) Columns:
<ul>
<li>chr1: Chromosome of the first anchor</li>
<li>start1: Start position of the first anchor</li>
<li>end1: End position of the first anchor</li>
<li>chr2: Chromosome of the second anchor</li>
<li>start2: Start position of the second anchor</li>
<li>end2: End position of the second anchor</li>
<li>dot: the dot .</li>
<li>couunt: raw loop counts</li>
</ul></li>
<li>Significant interaction files (*.interaction.Q0.01.txt) Columns:
<ul>
<li>chr1: Chromosome of the first anchor</li>
<li>start1: Start position of the first anchor</li>
<li>end1: End position of the first anchor</li>
<li>chr2: Chromosome of the second anchor</li>
<li>start2: Start position of the second anchor</li>
<li>end2: End position of the second anchor</li>
<li>count: raw loop counts</li>
<li>q_value: Adjuested P-value of the interaction</li>
</ul></li>
<li>WashU Genome Browser compatible tracks
(*.interaction.Q0.01.washu.txt.gz)</li>
</ol>
</div>
<div id="documentation" class="section level2">
<h2>Documentation</h2>
<p>For more detailed information about the functions and their usage,
you can use help in R:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a>?sintHiChIP</span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a>?run_sintHiChIP</span></code></pre></div>
<p>These will provide comprehensive documentation for the package and
its main function, including all parameters and their descriptions.</p>
</div>
<div id="workflow" class="section level2">
<h2>Workflow</h2>
<p>The workflow can be briefly described as below:</p>
<ol style="list-style-type: decimal">
<li>Data Preparation: Ensure you have the necessary input files (valid
pairs, peaks, etc.) ready.</li>
<li>Mode Selection: Choose between local and global mode based on your
analysis needs.</li>
<li>Run sintHiChIP: Use the run_sintHiChIP function with appropriate
parameters.</li>
<li>Output Analysis: Examine the output files for significant
interactions and other results.</li>
</ol>
</div>
<div id="tips-and-best-practices" class="section level2">
<h2>Tips and Best Practices</h2>
<p>To make better use of the sintHiChIP software, please keep the
following tips in mind:</p>
<ol style="list-style-type: decimal">
<li>Always use the latest version of sintHiChIP for the most up-to-date
features and bug fixes.</li>
<li>Carefully prepare your input files to ensure they are in the correct
format.</li>
<li>Adjust the FDR threshold based on your specific experimental needs
and the stringency required.</li>
<li>For large datasets, consider running sintHiChIP on a
high-performance computing cluster.</li>
<li>When running on a Linux or Unix system, ensure you have the
necessary permissions to execute the required software and access
input/output directories.</li>
</ol>
</div>
<div id="troubleshooting" class="section level2">
<h2>Troubleshooting</h2>
<p>If you encounter any issues while using sintHiChIP, consider the
following steps:</p>
<ol style="list-style-type: decimal">
<li>Check that all input files are correctly formatted and
accessible.</li>
<li>Ensure all required dependencies are installed and up to date.</li>
<li>Verify that you have sufficient disk space for output files.</li>
<li>For specific error messages, consult the package documentation or
seek help from the package maintainers.</li>
<li>If running on a non-Unix/Linux system, consider using a Linux/Unix
virtual machine or container to ensure compatibility.</li>
</ol>
</div>
<div id="conclusion" class="section level2">
<h2>Conclusion</h2>
<p>sintHiChIP provides a powerful and flexible tool for analyzing HiChIP
data. By offering both local and global modes, it caters to a wide range
of analysis needs. We hope this vignette helps you get started with
sintHiChIP and enables you to make the most of your HiChIP data
analysis.</p>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
